src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -1125,5 +1125,168 @@ public int GetMaxSignatureSize(DSASignatureFormat signatureFormat)
+        ///   <list type="bullet">
+        ///     <item><description>PUBLIC KEY</description></item>
+        ///     <item><description>PRIVATE KEY</description></item>
+        ///   </list>
+        ///   </para>
+        /// </remarks>
+        public override void ImportFromPem(ReadOnlySpan<char> input)
+        {
+            PemKeyImportHelpers.ImportPem(input, label => {
+                if (label.SequenceEqual("PRIVATE KEY"))
+                {
+                    return ImportPkcs8PrivateKey;
+                }
+                else if (label.SequenceEqual("PUBLIC KEY"))
+                {
+                    return ImportSubjectPublicKeyInfo;
+                }
+                else
+                {
+                    return null;
+                }
+            });
+        }
+
+        /// <summary>
+        /// Imports an encrypted RFC 7468 PEM-encoded private key, replacing the keys for this object.
+        /// </summary>
+        /// <param name="input">The PEM text of the encrypted key to import.</param>
+        /// <param name="password">
+        /// The password to use for decrypting the key material.
+        /// </param>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="input"/> does not contain a PEM-encoded key with a recognized label.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        /// <paramref name="input"/> contains multiple PEM-encoded keys with a recognized label.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///  The password is incorrect.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///  The base-64 decoded contents of the PEM text from <paramref name="input" />
+        ///  do not represent an ASN.1-BER-encoded PKCS#8 EncryptedPrivateKeyInfo structure.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///   The base-64 decoded contents of the PEM text from <paramref name="input" />
+        ///   indicate the key is for an algorithm other than the algorithm
+        ///   represented by this instance.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///   The base-64 decoded contents of the PEM text from <paramref name="input" />
+        ///   represent the key in a format that is not supported.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///   The algorithm-specific key import failed.
+        /// </exception>
+        /// <remarks>
+        ///   <para>
+        ///   When the base-64 decoded contents of <paramref name="input" /> indicate an algorithm that uses PBKDF1
+        ///   (Password-Based Key Derivation Function 1) or PBKDF2 (Password-Based Key Derivation Function 2),
+        ///   the password is converted to bytes via the UTF-8 encoding. This method
+        ///   only supports the binary (BER/CER/DER) encoding of EncryptedPrivateKeyInfo.
```

Comment:
I think that this method only supports the PEM encoded version of EncryptedPrivateKeyInfo :smile:.

Reply:
🤦‍♂

Reply:
Oh, I recall what I was trying to communicate - it just came out poorly. I was trying to say that the base-64 contents, when decoded, must be a BER, DER, CER {Encrypted}PrivateKeyInfo.